scripts.py

```diff
@@ -417,23 +418,48 @@ def do_run(self):
         work_ids_to_delete = set()
         unset_work_id = dict(work_id=None)
 
-        if self.force:
-            self.clear_existing_works()                  
+        if self.identifiers:
+            for i in self.identifiers:
+                pool = i.licensed_through
+                if not pool:
+                    logging.warn(
+                        "No LicensePool for %r, cannot create work.", i
+                    )
+                    continue
 
-        logging.info("Consolidating works.")
-        LicensePool.consolidate_works(self._db)
+                if pool.work:
+                    # We're about to delete a preexisting work. If the
+                    # problem is that this LicensePool is incorrectly
+                    # grouped together with some other LicensePool,
+                    # then that LicensePool must also have
+                    # calculate_work() called on it, so that we can
+                    # create two Works where there used to be one.
+                    all_pools = pool.work.license_pools
+                    self.clear_works(pool.work.id)
+                else:
+                    all_pools = [pool]
+                for pool in all_pools:
```

Comment:
And it also has ripple effects on other LicensePools associated (probably incorrectly) with this Work.